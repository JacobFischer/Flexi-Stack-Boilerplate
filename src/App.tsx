import React from "react";

export const App = () => <strong>Hello World!</strong>;
