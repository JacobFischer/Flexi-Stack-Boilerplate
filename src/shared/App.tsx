import React from "react";
import { Link, Route, Switch } from "react-router-dom";

export function App() {
    return (<>
        <ul>
            <li><Link to="/">Home</Link></li>
            <li><Link to="/about">About</Link></li>
            <li><Link to="/some-page">Some Page</Link></li>
            <li><Link to="/not-found">Dead link</Link></li>
        </ul>
        <Switch>
            <Route exact path="/" component={() => <p>Welcome to the index</p>} />
            <Route exact path="/about" component={() => <em>About the app</em>} />
            <Route exact path="/some-page" component={() => <strong>Some random other page</strong>} />
            <Route component={() => <pre>404 Not Found!</pre>} />
        </Switch>
    </>);
}
