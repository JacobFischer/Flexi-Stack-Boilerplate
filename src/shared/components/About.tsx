import React from "react";

export const About = (): JSX.Element => (
    <>
        <h2>About</h2>
        <p><strong>Jacob Fischer</strong> made this page! It&apos;s simple boilerplate to get up and moving ASAP.</p>
    </>
);
