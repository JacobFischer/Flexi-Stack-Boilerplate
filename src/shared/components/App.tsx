import React from "react";
import { Link, Route, Switch } from "react-router-dom";
import styled from "styled-components";
import { routes } from "../routes";
import { NotFound } from "./NotFound";

const NavBar = styled.ul({
    backgroundColor: "red",
});

export const App = () => (
    <>
        <NavBar>
            {routes.map(([route]) => (<li key={route}><Link to={route}>{route}</Link></li>))}
            <li><Link to="/not-found">Dead link</Link></li>
        </NavBar>
        <Switch>
            {routes.map(([route, component]) => <Route key={route} exact path={route} component={component} />)}
            <Route component={NotFound} />
        </Switch>
    </>
);
