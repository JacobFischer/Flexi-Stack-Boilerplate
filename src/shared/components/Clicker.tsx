import React from "react";
import styled from "styled-components";

export const Button = styled.button({
    backgroundColor: "lightblue",
    border: "3px solid blue",
});

/**
 * An example component that maintains a state via React hooks.
 *
 * @returns A functional component that uses state.
 */
export function Clicker() {
    // const [count, setCount] = useState(0);
    const count = -1;

    return (
        <div>
            <p>You clicked me {count} times</p>
            <Button onClick={undefined /*() => setCount(count + 1)*/}>
                Click me
            </Button>
        </div>
    );
}
export default Clicker;
