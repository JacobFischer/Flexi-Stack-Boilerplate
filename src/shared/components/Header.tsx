import React from 'react';
import styled from 'styled-components';

const DarkBlue = styled.h1({
  color: 'darkblue',
  fontSize: '2rem',
  fontWeight: 'bold',
});

/**
 * A simple main Header.
 *
 * @param props - Props to spread.
 * @returns A functional react component.
 */
const Header = <P extends unknown>(props: React.PropsWithChildren<P>) => (
  <DarkBlue {...props} />
);

export default Header;
