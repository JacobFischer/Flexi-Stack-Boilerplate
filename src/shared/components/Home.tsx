import React from "react";

export const Home = (): JSX.Element => (
    <>
        <h1>Home</h1>
        <p>Welcome to the index page!</p>
    </>
);
