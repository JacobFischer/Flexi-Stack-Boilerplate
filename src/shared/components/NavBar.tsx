import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

import { route as routeAbout } from '../pages/about/route';
import { route as routeHome } from '../pages/home/route';

const List = styled.ul({
  backgroundColor: 'darkgreen',
});

const Item = styled.li({
  color: 'white',
  display: 'inline',
  listStyle: 'none',
  padding: '0 0.5rem',
});

const NavBar: React.FunctionComponent = () => (
  <List>
    <Item>
      <Link to={routeHome}>Home</Link>
    </Item>
    <Item>
      <Link to={routeAbout}>About</Link>
    </Item>
    <Item>
      <Link to="/nowhere">Nowhere</Link>
    </Item>
  </List>
);

export default NavBar;
