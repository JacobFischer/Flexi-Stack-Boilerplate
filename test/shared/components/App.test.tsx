import React from "react";
import { StaticRouter } from "react-router";
import renderer from "react-test-renderer";
import { App } from "../../../src/shared/components/App";

const renderApp = (location?: string): renderer.ReactTestRendererJSON | null =>
    renderer
        .create(
            <StaticRouter location={location}>
                <App />
            </StaticRouter>,
        )
        .toJSON();

describe("App component", () => {
    it("renders", () => {
        expect(renderApp()).toMatchSnapshot();
    });

    it("renders the index Home page", () => {
        expect(renderApp("/")).toMatchSnapshot();
    });

    it("renders the About page", () => {
        expect(renderApp("/about")).toMatchSnapshot();
    });

    it("renders the SomePage page", () => {
        expect(renderApp("/some-page")).toMatchSnapshot();
    });

    it("renders the NotFound page on bad routes", () => {
        expect(renderApp("/hey-i-wont-exist")).toMatchSnapshot();
    });
});
