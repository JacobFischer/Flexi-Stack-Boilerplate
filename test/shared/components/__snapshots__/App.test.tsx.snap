// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders 1`] = `
Array [
  <ul
    className="App__NavBar-sc-24p3v9-0 jafctB"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        /
      </a>
    </li>
    <li>
      <a
        href="/about"
        onClick={[Function]}
      >
        /about
      </a>
    </li>
    <li>
      <a
        href="/clicker"
        onClick={[Function]}
      >
        /clicker
      </a>
    </li>
    <li>
      <a
        href="/some-page"
        onClick={[Function]}
      >
        /some-page
      </a>
    </li>
    <li>
      <a
        href="/not-found"
        onClick={[Function]}
      >
        Dead link
      </a>
    </li>
  </ul>,
  <div
    className="Home__Hot-sc-1bxbiux-0 jbQijg"
  >
    <h1>
      Home
    </h1>
    <p>
      Welcome to the index page!
    </p>
  </div>,
]
`;

exports[`App component renders the About page 1`] = `
Array [
  <ul
    className="App__NavBar-sc-24p3v9-0 jafctB"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        /
      </a>
    </li>
    <li>
      <a
        href="/about"
        onClick={[Function]}
      >
        /about
      </a>
    </li>
    <li>
      <a
        href="/clicker"
        onClick={[Function]}
      >
        /clicker
      </a>
    </li>
    <li>
      <a
        href="/some-page"
        onClick={[Function]}
      >
        /some-page
      </a>
    </li>
    <li>
      <a
        href="/not-found"
        onClick={[Function]}
      >
        Dead link
      </a>
    </li>
  </ul>,
  <h2>
    About
  </h2>,
  <p>
    Jacob Fischer
     made this page! It's simple boilerplate to get up and moving ASAP.
  </p>,
]
`;

exports[`App component renders the NotFound page on bad routes 1`] = `
Array [
  <ul
    className="App__NavBar-sc-24p3v9-0 jafctB"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        /
      </a>
    </li>
    <li>
      <a
        href="/about"
        onClick={[Function]}
      >
        /about
      </a>
    </li>
    <li>
      <a
        href="/clicker"
        onClick={[Function]}
      >
        /clicker
      </a>
    </li>
    <li>
      <a
        href="/some-page"
        onClick={[Function]}
      >
        /some-page
      </a>
    </li>
    <li>
      <a
        href="/not-found"
        onClick={[Function]}
      >
        Dead link
      </a>
    </li>
  </ul>,
  <h2>
    Not Found
  </h2>,
  <pre>
    Sorry, but we couldn't find anything here!
  </pre>,
]
`;

exports[`App component renders the SomePage page 1`] = `
Array [
  <ul
    className="App__NavBar-sc-24p3v9-0 jafctB"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        /
      </a>
    </li>
    <li>
      <a
        href="/about"
        onClick={[Function]}
      >
        /about
      </a>
    </li>
    <li>
      <a
        href="/clicker"
        onClick={[Function]}
      >
        /clicker
      </a>
    </li>
    <li>
      <a
        href="/some-page"
        onClick={[Function]}
      >
        /some-page
      </a>
    </li>
    <li>
      <a
        href="/not-found"
        onClick={[Function]}
      >
        Dead link
      </a>
    </li>
  </ul>,
  <h3>
    SomePage
  </h3>,
  <p>
    This is just some random page.
  </p>,
  <p>
    <em>
      I hope this gives you an idea of where to go with this project!
    </em>
  </p>,
]
`;

exports[`App component renders the index Home page 1`] = `
Array [
  <ul
    className="App__NavBar-sc-24p3v9-0 jafctB"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        /
      </a>
    </li>
    <li>
      <a
        href="/about"
        onClick={[Function]}
      >
        /about
      </a>
    </li>
    <li>
      <a
        href="/clicker"
        onClick={[Function]}
      >
        /clicker
      </a>
    </li>
    <li>
      <a
        href="/some-page"
        onClick={[Function]}
      >
        /some-page
      </a>
    </li>
    <li>
      <a
        href="/not-found"
        onClick={[Function]}
      >
        Dead link
      </a>
    </li>
  </ul>,
  <div
    className="Home__Hot-sc-1bxbiux-0 jbQijg"
  >
    <h1>
      Home
    </h1>
    <p>
      Welcome to the index page!
    </p>
  </div>,
]
`;
